labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +59,15 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case s:String => "A string with length " + s.length()
+      case i:Int if i > 0 => "A positive integer"
+      case p:Person => "A person with name: " + p.name
+      case seq:Seq[Any] if seq.size > 10 => "Seq with more than 10 elements"
+      case seq:Seq[Any] if seq.size > 2 => "first: " + seq.head + ", second: " + seq.apply(1) + ", rest: " + seq.drop(2)
```

Comment:
[Scala string interpolation](http://docs.scala-lang.org/overviews/core/string-interpolation.html) can be a nice way to build strings, without having to use `+`.